src/components/profile/EditAvatar.js

```diff
@@ -41,11 +41,13 @@ const EditAvatar = ({ theme, screenProps }) => {
 
       log.error('saving image failed:', message, exception, { dialogShown: true })
       showErrorDialog('We could not capture all your beauty. Please try again.')
-    } finally {
-      setProcessing(false)
     }
   }, [screenProps, markAsDirty, setProcessing, showErrorDialog, user])
 
+  useEffect(() => {
```

Comment:
why do we need this as an effect?

Reply:
This is the easiest way to call setProcessing(false) in the end without triggering any warnings or causing memory leaks

Reply:
not agree with that
finally and promise's .finally are giving 100% guarantee this code block will be executed despite of exception/rejection status

please revert this change